src/language/patterns.md

```diff
@@ -0,0 +1,334 @@
+---
+title: Patterns
+description: Summary of patterns in Dart.
+---
+
+{{site.alert.version-note}}
+  Patterns require a [language version][] of at least 3.0.
+{{site.alert.end}}
+
+Patterns are a syntactic category in the Dart language, like statements and expressions.
+They represent the form of a value, not an actual value.
```

Comment:
Maybe "shape of values"?

Reply:
I may be wrong, but I think of a constant pattern as an "actual value"

Reply:
I think of it as a constant pattern *has* a value for the constant that it tests, just like a relational pattern has a value for the right-hand side operand. But I don't think of patterns as being value-like in general.

The "set of values" and "actual values" feels like a confusing level of repetition. The only "set of values" the pattern represents are the actual ones it would match.

If you look up "pattern" in the dictionary, the two definitions that are closest are:

> * an arrangement or sequence regularly found in comparable objects. "the house had been built on the usual pattern"

> * a regular and intelligible form or sequence discernible in certain actions or situations. "a complicating factor is the change in working patterns"

I think, "regular", "comparable", and "discernable" are the key bits.

A pattern describes some commonality that a set of objects share and you can determine whether a given object has that commonality or not. A pattern is like a mold, gauge, or yardstick that you can hold an object up to and see if it fits.

So maybe, "A pattern describes a set of properties an object might have. A value can be *matched* against the pattern to see if it has those properties."